OpenRA.Mods.Common/Traits/AppearsOnRadar.cs

```diff
@@ -42,6 +42,14 @@ public void Created(Actor self)
 		{
 			var color = modifier != null ? modifier.RadarColorOverride(self) : self.Owner.Color.RGB;
 
+			if (Game.Settings.Game.UsePlayerStanceColors)
+			{
+				color = self.Owner.PlayerStanceColor(self, modifier);
+
+				if (color == Color.Tan)
```

Comment:
Handling the modifier as suggested above may also make this workaround unnecessary.  The civilian actors that don't implement RadarColorFromTerrain probably *do* want to be shown as tan.